content/en/blog/_posts/2019-05-22-OPA-Gatekeeper-Policy-and-Governance-for-Kubernetes.md

```diff
@@ -0,0 +1,194 @@
+
+```yaml
+apiVersion: constraints.gatekeeper.sh/v1beta1
+kind: K8sRequiredLabels
+metadata:
+  name: ns-must-have-hr
+spec:
+  match:
+    kinds:
+      - apiGroups: [""]
+        kinds: ["Namespace"]
+  parameters:
+    labels: ["hr"]
+status:
+  auditTimestamp: "2019-08-03T01:46:13Z"
+  byPod:
+  - enforced: true
+    id: gatekeeper-controller-manager-0
+  violations:
+  - enforcementAction: deny
+    kind: Namespace
+    message: 'you must provide labels: {"hr"}'
+    name: default
+  - enforcementAction: deny
+    kind: Namespace
+    message: 'you must provide labels: {"hr"}'
+    name: gatekeeper-system
+  - enforcementAction: deny
+    kind: Namespace
+    message: 'you must provide labels: {"hr"}'
+    name: kube-public
+  - enforcementAction: deny
+    kind: Namespace
+    message: 'you must provide labels: {"hr"}'
+    name: kube-system
+```
+
+### Data Replication 
+
+Audit requires replication of Kubernetes resources into OPA before they can be evaluated against the enforced Constraints. Data replication is also required by Constraints that need access to objects in the cluster other than the object under evaluation. For example, a Constraint that enforces uniqueness of ingress hostname must have access to all other ingresses in the cluster. 
+
+To configure Kubernetes data to be replicated, create a sync config resource with the resources to be replicated into OPA. For example, the below configuration replicates all namespace and pod resources to OPA. 
+
+```yaml
+apiVersion: config.gatekeeper.sh/v1alpha1
+kind: Config
+metadata:
+  name: config
+  namespace: "gatekeeper-system"
+spec:
+  sync:
+    syncOnly:
+      - group: ""
+        version: "v1"
+        kind: "Namespace"
+      - group: ""
+        version: "v1"
+        kind: "Pod"
+```
+
+## Planned for Future 
+
+The community behind the Gatekeeper project will be focusing on providing mutating admission control to support mutation scenarios (for example: annotate objects automatically with departmental information when creating a new resource), support external data to inject context external to the cluster into the admission decisions, support dry run to see impact of a policy on existing resources in the cluster before enforcing it, and more audit functionalities.  
+
+If you are interested in learning more about the project, check out the [Gatekeeper](https://github.com/open-policy-agent/gatekeeper) repo. If you are interested to help define the direction of Gatekeeper, join the [#kubernetes-policy](https://openpolicyagent.slack.com/messages/CDTN970AX) channel on OPA Slack, and join our [weekly meetings](https://docs.google.com/document/d/1A1-Q-1OMw3QODs1wT6eqfLTagcGmgzAJAjJihiO3T48/edit) to discuss development, issues, use cases, etc.  
```

Comment:
**Nitpick**: `If you are interested to help define the direction of Gatekeeper...` could be changed to say `If you are interested in helping define the direction of Gatekeeper...`